agent/consul/server.go

```diff
@@ -462,6 +466,13 @@ func NewServer(config *Config, flat Deps, externalGRPCServer *grpc.Server) (*Ser
 		Logger:   logger.Named("hcp_manager"),
 	})
 
+	// TODO(NET-1380, NET-1381): thread this into the net/rpc and gRPC interceptors.
+	s.incomingRPCLimiter = rpcRate.NewHandler(rpcRate.HandlerConfig{
```

Comment:
I'm not sure this is the right place for grpc, as at that level the grpc server is already created and we need to add the interceptor at the creation.

Reply:
It's a little awkward for sure. An alternative would be to put it on `BaseDeps` which doesn't feel right either, as we'll never actually use it on client agents.

Here's how I'm using it in my draft gRPC PR:

https://github.com/hashicorp/consul/blob/c6c6ec999d06263e6c997afa14d8b955add9297a/agent/agent.go#L565-L578

https://github.com/hashicorp/consul/blob/c6c6ec999d06263e6c997afa14d8b955add9297a/agent/consul/server.go#L1830-L1833